internal/slicer/slicer_test.go

```diff
@@ -769,6 +769,79 @@ var slicerTests = []slicerTest{{
 	report: map[string]string{
 		"/dir/nested/file": "file 0644 84237a05 {test-package_myslice}",
 	},
+}, {
+	summary: "Generate (chisel-state) does not impact the fs",
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/file:
+						/dir/file-copy:  {copy: /dir/file}
+						/other-dir/file: {symlink: ../dir/file}
+						/dir/text-file:  {text: data1}
+						/dir/foo/bar/:   {make: true, mode: 01777}
+						/var/lib/foo/**: {generate: chisel-state}
+		`,
+	},
+	filesystem: map[string]string{
+		"/dir/":           "dir 0755",
+		"/dir/file":       "file 0644 cc55e2ec",
+		"/dir/file-copy":  "file 0644 cc55e2ec",
+		"/dir/foo/":       "dir 0755",
+		"/dir/foo/bar/":   "dir 01777",
+		"/dir/text-file":  "file 0644 5b41362b",
+		"/other-dir/":     "dir 0755",
+		"/other-dir/file": "symlink ../dir/file",
+	},
+	report: map[string]string{
+		"/dir/file":       "file 0644 cc55e2ec {test-package_myslice}",
+		"/dir/file-copy":  "file 0644 cc55e2ec {test-package_myslice}",
+		"/dir/foo/bar/":   "dir 01777 {test-package_myslice}",
+		"/dir/text-file":  "file 0644 5b41362b {test-package_myslice}",
+		"/other-dir/file": "symlink ../dir/file {test-package_myslice}",
+	},
+}, {
+	summary: "Generate (chisel-state) directory is not extracted from package if exists",
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/nested/**: {generate: chisel-state}
+		`,
+	},
+	filesystem: map[string]string{},
+	report:     map[string]string{},
+}, {
+	summary: "Slice with generate: chisel-state is installed always",
```

Comment:
I dont think we want to install chisel-state if the slice is not selected. I cannot see any mention of it in the spec but I would assume we want to treat it as a regular slice (we only make the exception of installing base-files automatically which is precisely why the db is going to be there).

Reply:
Agreed. Removed in https://github.com/letFunny/chisel/pull/11/commits/bf534c0c8a03c7cbe68a21f35bfedfbe1eeac40d.